src/components/pages/__tests__/BeregnPage.test.tsx

```diff
@@ -0,0 +1,289 @@
+import React from 'react'
+import { act, render, screen, waitFor } from '@testing-library/react'
+import { FormContext } from '@/contexts/context'
+import BeregnPage from '../BeregnPage'
+import { initialFormState } from '@/defaults/defaultFormState'
+import { fetchBeregnData } from '../BeregnPage'
+import { produce } from 'immer'
+import submitForm from '@/functions/submitForm'
+
+jest.mock('../BeregnPage', () => {
+  const originalModule = jest.requireActual('../BeregnPage')
+  return {
+    __esModule: true,
+    ...originalModule,
+    fetchBeregnData: jest.fn(),
+  }
+})
+
+jest.mock('@/functions/submitForm', () => jest.fn())
+
+const mockFetchBeregnData = fetchBeregnData as jest.Mock
+const mockSubmitForm = submitForm as jest.Mock
+
+const mockBeregnResult = {
+  alderspensjon: [
+    { alder: 67, beloep: 200000 },
+    { alder: 68, beloep: 210000 },
+  ],
+  afpPrivat: [
+    { alder: 67, beloep: 50000 },
+    { alder: 68, beloep: 55000 },
+  ],
+  afpOffentlig: [],
+  vilkaarsproeving: {
+    vilkaarErOppfylt: true,
+  },
+}
+
+const mockGoToNext = jest.fn()
+const mockSetState = jest.fn()
+
+const defaultFormPageProps = {
+  curStep: 1,
+  length: 5,
+  goBack: jest.fn(),
+  onStepChange: jest.fn(),
+  handleSubmit: jest.fn(),
+  goToNext: mockGoToNext,
+}
+
+const mockContextValue = {
+  state: initialFormState,
+  setState: mockSetState,
+  formPageProps: defaultFormPageProps,
+}
+
+describe('BeregnPage Component', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    mockSubmitForm.mockResolvedValue(JSON.stringify(mockBeregnResult))
+    mockFetchBeregnData.mockImplementation(() => {
+      return {
+        read: () => mockSubmitForm(),
+      }
+    })
+  })
+
+  describe('Rendring', () => {
+    test('Burde rendre Beregn komponenten', async () => {
+      await act(async () => {
+        render(
+          <FormContext.Provider value={mockContextValue}>
+            <BeregnPage />
+          </FormContext.Provider>
+        )
+      })
+
+      await waitFor(() => {
+        expect(screen.getByRole('region')).toBeInTheDocument()
+      })
+
+      const beregnElement = screen.getByText('Resultat')
+      expect(beregnElement).toBeInTheDocument()
+      const boxElement = screen.getByRole('region')
+      expect(boxElement).toBeInTheDocument()
+    })
+  })
+
+  describe('Gitt at noen states behøver å oppdateres', () => {
+    test('Burde Immer sin produce funksjon kunne oppdatere statene som forventet', async () => {
+      await act(async () => {
+        render(
+          <FormContext.Provider value={mockContextValue}>
+            <BeregnPage />
+          </FormContext.Provider>
+        )
+      })
+
+      await waitFor(() => {
+        expect(screen.getByText('Resultat')).toBeInTheDocument()
+      })
+
+      const expectedState = produce(initialFormState, (draft) => {
```

Comment:
Her tester du immer sin funksjon. Det gir ikke verdi for vår kodebase.
Det jeg tror du vil teste er at:
- BeregnPage lastes med en gitt payload i staten fra før av
- sjekke at payloadden oppdateres ift de reglene som er kodet i din useEffect

Det kan se ut som at det allerede er det du tester under

Reply:
Det tester jeg under ja. Ønsker du at jeg kune fjerner følgende test: 'Burde Immer sin produce funksjon kunne oppdatere statene som forventet'

Reply:
Det kan se ut til meg som at den bare tester immer sin funksjon. Ta en titt i coverage om den dekker noe som de andre testene ikke allerede dekker. Gjør den ikke det, kan den fjernes.